src/main/java/seedu/address/model/person/ClientStatus.java

```diff
@@ -35,25 +35,55 @@ public String toString() {
 
     public final Status value;
 
+    /**
+     * Constructs a {@code ClientStatus}.
+     * Only to be used when reading Json files and by other methods in the {@code ClientStatus} class.
+     *
+     * @param status The ordinal of the status enumeration.
+     */
     public ClientStatus(int status) {
         value = Status.values()[status];
     }
+
+    /**
+     * Returns a {@code ClientStatus} for non-client persons.
+     * Used only when adding a non-client person to the addressbook.
+     */
     public static ClientStatus initNotClientStatus() {
         return new ClientStatus(0);
     }
 
+    /**
+     * Returns a {@code ClientStatus} for client persons.
+     * Used only when adding a client person to the addressbook.
+     */
     public static ClientStatus initClientStatus() {
         return new ClientStatus(1);
     }
 
+    /**
+     * Returns the ordinal of the {@code ClientStatus}'s value in the Status enumeration.
+     */
     public int getStatus() {
         return value.ordinal();
     }
 
+    /**
+     * Increases the value of {@code ClientStatus} by one stage according to the Status enumeration.
+     * The highest value is {@code END}.
+     *
+     * @return A new instance of {@code ClientStatus} that has a higher value up to the maximum value.
+     */
     public ClientStatus increment() {
         return new ClientStatus(Math.min(this.getStatus() + 1, CLIENT_MAX));
     }
 
```

Comment:
Good use of OOP structure in using getter methods to access Status as well as abstracted out static constants like CLIENT_MAX